marconi-chain-index/src/Marconi/ChainIndex/Indexers/Utxo.hs

```diff
@@ -215,93 +225,81 @@ instance ToJSON Utxo where
             , "txIndexInBlock"    .= (u ^. txIndexInBlock)
             ]
 
-data ChainPointRow
-    = ChainPointRow { _cpSlotNo :: C.SlotNo, _cpBlockHash :: C.Hash C.BlockHeader }
-    deriving (Show, Eq, Ord, Generic)
-
-$(makeLenses ''ChainPointRow)
-
-toChainPointRow :: C.ChainPoint -> Maybe ChainPointRow
-toChainPointRow cp = ChainPointRow <$> C.chainPointToSlotNo cp <*> C.chainPointToHeaderHash cp
-
-instance FromJSON ChainPointRow where
-    parseJSON (Object v)
-        = ChainPointRow
-            <$> v .: "slotNo"
-            <*> v .: "blockHeaderHash"
-    parseJSON _ = mempty
-
-instance ToJSON ChainPointRow where
-    toJSON c
-        = object
-        [ "slotNo" .= view cpSlotNo c
-        , "blockHeaderHash" .= view cpBlockHash c
-        ]
-
-getChainPoint :: ChainPointRow -> C.ChainPoint
-getChainPoint cp = C.ChainPoint (cp ^. cpSlotNo) (cp ^. cpBlockHash)
-
 data SpentInfo
     = SpentInfo
-    { _siSpentPoint :: ChainPointRow
-    , _siSpentTxId  :: C.TxId
+    { _siSlotNo          :: C.SlotNo
+    , _siBlockHeaderHash :: C.Hash C.BlockHeader
+    , _siSpentTxId       :: C.TxId
     } deriving (Show, Eq, Ord, Generic)
 
 $(makeLenses ''SpentInfo)
 
 data UtxoRow = UtxoRow
-  { _urUtxo          :: !Utxo
-  , _urCreationPoint :: !ChainPointRow
-  , _urSpentInfo     :: !(Maybe SpentInfo)
+  { _urUtxo            :: !Utxo
+  , _urSlotNo          :: !C.SlotNo
+  , _urBlockHeaderHash :: !(C.Hash C.BlockHeader)
+  , _urSpentInfo       :: !(Maybe SpentInfo)
   } deriving (Show, Eq, Ord, Generic)
 
 $(makeLenses ''UtxoRow)
 
-urSpentSlotNo :: Traversal' UtxoRow C.SlotNo
-urSpentSlotNo = urSpentInfo . _Just . siSpentPoint . cpSlotNo
+data UtxoResult = UtxoResult_
```

Comment:
UtxoResult_? Why not just `UtxoResult`?

Reply:
The `UtxoResult` was taken by the query.. Maybe another name to avoid the underscore entirely?

Reply:
Also, why define `UtxoResult` vs use the already existin `UtxoRow`?

Reply:
After the rebase I've forgotten to remove the datum field from `Utxo` so that is why `UtxoRow` and `UtxoResult` are still isomorphic.. But the idea is for talking to sqlite to have different data types for incoming and outgoing data: when writing to the database then there are two[1] tables: `unspent_transactions` and `datumhash_datum`, which both should have their own row types (`UtxoRow` and `DatumRow`). And then there is the sql select that is used to reply to indexer queries which has all the tables joined, this is the `UtxoResult`.

[1] Actually, `spent` is now back too, but adding this could be another jira ticket.

Reply:
Ahh I understand now. In that case, yes, your solution makes sense.